File: ui/lib/api/demo/media.dart
import 'package:mizer/api/contracts/media.dart';
import 'package:mizer/protos/media.pb.dart';

class MediaDemoApi implements MediaApi {
  @override
  Future<MediaTag> createTag(String name) {
    // TODO: implement createTag
    throw UnimplementedError();
  }

  @override
  Future<MediaFiles> getMedia() async {
    return MediaFiles();
  }

  @override
  Future<GroupedMediaFiles> getTagsWithMedia() async {
    return GroupedMediaFiles();
  }

}

File: api/src/handlers/media.rs
use mizer_media::MediaServer;

use crate::models::media::*;

#[derive(Clone)]
pub struct MediaHandler {
    api: MediaServer,
}

impl MediaHandler {
    pub fn new(api: MediaServer) -> Self {
        Self { api }
    }

    #[tracing::instrument(skip(self))]
    pub fn create_tag(&self, name: String) -> anyhow::Result<MediaTag> {
        let document = self.api.create_tag(name)?;

        Ok(document.into())
    }

    #[tracing::instrument(skip(self))]
    pub fn get_tags_with_media(&self) -> anyhow::Result<GroupedMediaFiles> {
        let tags = self.api.get_tags()?;
        let tags = tags.into_iter().map(MediaTagWithFiles::from).collect();

        Ok(GroupedMediaFiles {
            tags,
            ..Default::default()
        })
    }

    #[tracing::instrument(skip(self))]
    pub fn get_media(&self) -> anyhow::Result<MediaFiles> {
        let files = self.api.get_media()?;
        let files = files.into_iter().map(MediaFile::from).collect();

        Ok(MediaFiles {
            files,
            ..Default::default()
        })
    }
}
